From b2a6c933645e05c8d68ab01299dcbeb84f0faa08 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Mon, 3 Feb 2014 13:46:59 -0800
Subject: [PATCH 517/974] netdev: octeon3-ethernet: Try to set BGX mode
 correctly.

Still not working.

Signed-off-by: David Daney <david.daney@cavium.com>
[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/net/ethernet/octeon/octeon-bgx-port.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/octeon/octeon-bgx-port.c b/drivers/net/ethernet/octeon/octeon-bgx-port.c
index 87c8f38..efa8ece 100644
--- a/drivers/net/ethernet/octeon/octeon-bgx-port.c
+++ b/drivers/net/ethernet/octeon/octeon-bgx-port.c
@@ -47,6 +47,7 @@ struct bgx_port_priv {
 	int bgx_interface;
 	int index; /* Port index on BGX block*/
 	int ipd_port;
+	int xiface;
 	const u8 *mac_addr;
 	struct phy_device *phydev;
 	struct device_node *phy_np;
@@ -200,7 +201,9 @@ int bgx_port_enable(struct net_device *netdev)
 {
 	struct bgx_port_priv *priv = bgx_port_netdev2priv(netdev);
 
+	cvmx_helper_set_1000x_mode(priv->xiface, priv->index, false);
 	if (priv->phy_np == NULL) {
+		cvmx_helper_set_mac_phy_mode(priv->xiface, priv->index, false);
 		netif_carrier_on(netdev);
 		return 0;
 	}
@@ -210,6 +213,7 @@ int bgx_port_enable(struct net_device *netdev)
 	if (!priv->phydev)
 		return -ENODEV;
 
+	cvmx_helper_set_mac_phy_mode(priv->xiface, priv->index, true);
 	netif_carrier_off(netdev);
 	phy_start_aneg(priv->phydev);
 
@@ -231,7 +235,6 @@ static int bgx_port_probe(struct platform_device *pdev)
 	u32 index;
 	int r;
 	struct bgx_port_priv *priv;
-	int xiface;
 	int numa_node;
 
 	reg = of_get_property(pdev->dev.parent->of_node, "reg", NULL);
@@ -251,8 +254,8 @@ static int bgx_port_probe(struct platform_device *pdev)
 	priv->numa_node = numa_node;
 	priv->bgx_interface = (addr >> 24) & 0xf;
 	priv->index = index;
-	xiface = cvmx_helper_node_interface_to_xiface(numa_node, priv->bgx_interface);
-	priv->ipd_port = cvmx_helper_get_ipd_port(xiface, index);
+	priv->xiface = cvmx_helper_node_interface_to_xiface(numa_node, priv->bgx_interface);
+	priv->ipd_port = cvmx_helper_get_ipd_port(priv->xiface, index);
 	if (mac)
 		priv->mac_addr = mac;
 
-- 
2.6.2

